misc/bazel/lfs.bzl

```diff
@@ -0,0 +1,88 @@
+def lfs_smudge(repository_ctx, srcs, extract = False, stripPrefix = None):
+    for src in srcs:
+        repository_ctx.watch(src)
+    script = Label("//misc/bazel/internal:git_lfs_probe.py")
+    python = repository_ctx.which("python3") or repository_ctx.which("python")
```

Comment:
Wouldn't `python3` be enough?
Also: Shouldn't we rely on the bazel python toolchain, as an implicit dependency here instead of calling `which`? That way, once we ship a hermetic toolchain, we don't need to change this code. I believe that's done in other places where we invoke python as part of the build.

Reply:
Ah yes, the toolchain resolution mechanism is probably indeed unavailable. We could rely on the defined repo from [this](https://rules-python.readthedocs.io/en/latest/toolchains.html#other-toolchain-details) though, and reference it via a (private) attribute - that ought to order the repository resolution order in a way that we first fetch python, and then the lfs files.

Reply:
Tried it, but didn't have any luck. You can indeed do
```
python = use_extension("@rules_python//python/extensions:python.bzl", "python")
python.toolchain(
    python_version = "3.11",
)
use_repo(python, python3 = "python_3_11")
```
and have `@python3` point to the toolchain interpreter. But that will only work for normal rules, not repository ones (it's actually an `alias` pointing to, for example, `@python_3_11_x86_64-unknown-linux-gnu//:python3`, and this kind of targets do not work in the context of repository rules.

Reply:
exposing `python_3_11_x86_64-unknown-linux-gnu` with `python.use_repo` and using `python_3_11_x86_64-unknown-linux-gnu//:python` as a hidden attribute works, but that's obviously not doable in a cross-platform way :shrug: